test/builtin_dataset_mocks.py

```diff
@@ -1329,20 +1329,20 @@ def cub200(info, root, config):
 
 @register_mock
 def eurosat(info, root, config):
-    data_folder = pathlib.Path(root, "eurosat", "2750")
+    data_folder = pathlib.Path(root, "2750")
     data_folder.mkdir(parents=True)
 
     num_examples_per_class = 3
-    classes = ("AnnualCrop", "Forest")
-    for cls in classes:
+    categories = ["AnnualCrop", "Forest"]
+    for category in categories:
         create_image_folder(
             root=data_folder,
-            name=cls,
-            file_name_fn=lambda idx: f"{cls}_{idx}.jpg",
+            name=category,
+            file_name_fn=lambda idx: f"{category}_{idx + 1}.jpg",
             num_examples=num_examples_per_class,
         )
     make_zip(root, "EuroSAT.zip", data_folder)
```

Comment:
By using `data_folder` here, it is assumed that this folder and all of its contents should be included in the archive. By default, `make_zip` removes all files and folders it includes. Since the extra `eurosat` folder is not included, it is left as is. Thus, there are now an empty `eurosat` folder as well as the `EuroSAT.zip` file in the root directory.

Although not correct, that wouldn't be an issue if weren't for a quirk on Windows and macOS: `pathlib` treats file and folder names case-insensitive :exploding_head: This means, 

https://github.com/pytorch/vision/blob/e836b3d8baf8cbfdd18fcc872631962345a65689/torchvision/prototype/datasets/utils/_resource.py#L92-L99

is picking up the empty `eurosat` folder as a candidate for `EuroSAT.zip` although it should only look for an `EuroSAT` folder. Since we prefer extracted files over archives, it now loads the empty folder and thus all tests fail.

Reply:
> Windows and macOS: pathlib treats file and folder names case-insensitive

Is this a bug, or a documented "feature" ? Are we likely to hit more bugs in the future because of this?